lib/ansible/module_utils/urls.py

```diff
@@ -300,6 +302,51 @@ def http_open(self, req):
         return self.do_open(UnixHTTPConnection(self._unix_socket), req)
 
 
+class ProxyHandler(urllib.request.ProxyHandler):
+    _SPLITPORT_RE = re.compile('(.*):([0-9]*)', re.DOTALL)
+
+    @classmethod
+    def _splitport(cls, host):
+        # Copied from cpython urllib.parse
+        match = cls._SPLITPORT_RE.fullmatch(host)
+        if match:
+            host, port = match.groups()
+            if port:
+                return host, port
+        return host, None
```

Comment:
Perhaps we could simplify this to have a single obvious `return` statement?
```suggestion
        # Derived from cpython urllib.parse
        port = None
        if (match := cls._SPLITPORT_RE.fullmatch(host)):
            host, port = match.groups()
        return host, port or None
```